tests/test_visitation.py

```diff
@@ -302,5 +306,101 @@ def _get_execution_output(self, arn):
         return execution['output']
 
 
+@testmode.standalone
+class TestConsistencyVisitation(unittest.TestCase):
+    # Number of keys returned per round-trip by the mock blobstore's list_v2 iterator
+    page_size = 100
+    # Number of bundles in each mock replica
+    num_keys = 9 * page_size + random.randint(-page_size, page_size)
+    replicas = [r for r in Replica]
+    # Distance (in # of keys) between missing keys in each replica
+    key_drop_period = random.randint(10, 19)
+    # How much time to give the walker for actually processing the bucket listings
+    timeout = .1
+    # Minimum number of walk() invocations per replica (assuming one worker)
+    num_key_partitions = len(StorageVisitation.prefix_chars)
+
+    def setUp(self):
+        keys = sorted(f'bundles/{uuid.uuid4()}' for _ in range(self.num_keys))
+        # Drop every n-th key from either replica, but don't drop a key from all replicas
+        self.assertTrue(len(self.replicas) < self.key_drop_period)
+        self.keys = {replica: [key for key in keys if hash(key) % self.key_drop_period != i]
+                     for i, replica in enumerate(self.replicas)}
+        self.blobstore_mocks = []
+
+    @mock.patch('dss.util.aws.clients.stepfunctions.start_execution')
+    @mock.patch('dss.config.Config.get_blobstore_handle')
+    def test(self, get_blobstore_handle, start_execution):
+        start_execution.side_effect = self._start_execution
+        get_blobstore_handle.side_effect = self._get_blobstore_handle
+        visitation = StorageVisitation.start(number_of_workers=1,
+                                             replicas=[replica.name for replica in self.replicas],
+                                             buckets=[None] * len(self.replicas))
+        # Assert the magic value returned from _start_execution to prove that it was invoked
+        self.assertEquals(2 * self.num_keys - sum(map(len, self.keys.values())), int(visitation['arn']))
```

Comment:
Why 2?

Reply:
Great catch. That should be `len(self.replicas)`.

Reply:
Fixed.